qadence_libs/constructors/ufa_models/config.py

```diff
@@ -0,0 +1,284 @@
+
+        if self.inputs is None:
+            self.inputs = [f"phi_{i}" for i in range(self.num_features)]
+
+
+@dataclass
+class AnsatzConfig:
+    num_layers: int
+    """Number of layers of the ansatz."""
+
+    ansatz_type: str
+    """What type of ansatz.
+
+    "hea" for Hardware Efficient Ansatz.
+    "iia" for Identity intialized Ansatz.
+    """
+
+    ansatz_strategy: str
+    """Ansatz strategy.
+
+    "digital" for fully digital ansatz. Required if `ansatz_type` is `iia`.
+    "sdaqc" for analog entangling block.
+    "rydberg" for fully rydberg hea ansatz.
+    """
+
+    strategy_args: dict = field(default_factory=dict)
+    """
+    A dictionary containing keyword arguments to the function creating the ansatz.
+
+    Details about each below.
+
+    For "digital" strategy, accepts the following:
+        periodic (bool): if the qubits should be linked periodically.
+            periodic=False is not supported in emu-c.
+        operations (list): list of operations to cycle through in the
+            digital single-qubit rotations of each layer.
+            Defaults to  [RX, RY, RX] for hea and [RX, RY] for iia.
+        entangler (AbstractBlock): 2-qubit entangling operation.
+            Supports CNOT, CZ, CRX, CRY, CRZ, CPHASE. Controlld rotations
+            will have variational parameters on the rotation angles.
+            Defaults to CNOT
+
+    For "sdaqc" strategy, accepts the following:
+        operations (list): list of operations to cycle through in the
+            digital single-qubit rotations of each layer.
+            Defaults to  [RX, RY, RX] for hea and [RX, RY] for iia.
+        entangler (AbstractBlock): Hamiltonian generator for the
+            analog entangling layer. Time parameter is considered variational.
+            Defaults to NN interaction.
+
+    For "rydberg" strategy, accepts the following:
+        addressable_detuning: whether to turn on the trainable semi-local addressing pattern
+            on the detuning (n_i terms in the Hamiltonian).
+            Defaults to True.
+        addressable_drive: whether to turn on the trainable semi-local addressing pattern
+            on the drive (sigma_i^x terms in the Hamiltonian).
+            Defaults to False.
+        tunable_phase: whether to have a tunable phase to get both sigma^x and sigma^y rotations
+            in the drive term. If False, only a sigma^x term will be included in the drive part
+            of the Hamiltonian generator.
+            Defaults to False.
+    """
+    # The default for a dataclass can not be a mutable object without using this default_factory.
+
+    param_prefix: str = "theta"
+    """The base bame of the variational parameter."""
+
+    def __post_init__(self) -> None:
+        if self.ansatz_type == "iia":
+            assert (
+                self.ansatz_strategy != "rydberg"
+            ), "Rydberg strategy not allowed for Identity-initialized ansatz."
+
+
+@dataclass
+class ObservableConfig:
```

Comment:
hey @smitchaudhary , lets add the logic for output transformation here

Reply:
Yeah! It can already do a scaling operation for each qubit. Will expand that to include shifting too. Also I need to make the names a bit more clear.